pkg/tsdb/provider_proxy.go

```diff
@@ -0,0 +1,132 @@
+package tsdb
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin/coreplugin"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin/pluginextensionv2"
+)
+
+type providerProxy struct {
+	logger    log.Logger
+	providers map[string]coreplugin.ServeOpts
+}
+
+func newProviderProxy(logger log.Logger) *providerProxy {
+	return &providerProxy{
+		logger:    logger,
+		providers: map[string]coreplugin.ServeOpts{},
+	}
+}
+
+func (p *providerProxy) register(providerID string, opts coreplugin.ServeOpts) {
+	p.providers[providerID] = opts
+}
+
+func (p *providerProxy) ConfigureProvider(ctx context.Context, req *pluginextensionv2.ConfigureProviderRequest) (*pluginextensionv2.ConfigureProviderResponse, error) {
+	providers := []string{}
+	for provider := range p.providers {
+		providers = append(providers, provider)
+	}
+
+	return &pluginextensionv2.ConfigureProviderResponse{
+		Provider: providers,
+	}, nil
+}
+
+func (p *providerProxy) getProvider(providerID string) (coreplugin.ServeOpts, bool) {
+	opts, exists := p.providers[providerID]
+	return opts, exists
+}
+
+func (p *providerProxy) CollectMetrics(ctx context.Context) (*backend.CollectMetricsResult, error) {
+	return nil, backendplugin.ErrMethodNotImplemented
+}
+
+func (p *providerProxy) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+	providerID, err := backendplugin.ProviderIDFromContext(ctx)
```

Comment:
See other comment regarding why this is a bit of a hack currently and why we currently need it.